iothub_service_client/src/iothub_service_client_auth.c

```diff
@@ -191,13 +253,21 @@ IOTHUB_SERVICE_CLIENT_AUTH_HANDLE IoTHubServiceClientAuth_CreateFromConnectionSt
                         result = NULL;
                     }
                     /*Codes_SRS_IOTHUBSERVICECLIENT_12_028: [** IoTHubServiceClientAuth_CreateFromConnectionString shall allocate memory and copy sharedAccessKey to result->sharedAccessKey by calling mallocAndStrcpy_s. **] */
-                    else if (mallocAndStrcpy_s(&result->sharedAccessKey, sharedAccessKey) != 0)
+                    else if (!useSharedAccessSignature && (sharedAccess != NULL) && mallocAndStrcpy_s(&result->sharedAccessKey, sharedAccess) != 0)
                     {
                         /*Codes_SRS_IOTHUBSERVICECLIENT_12_029: [** If the mallocAndStrcpy_s fails, IoTHubServiceClientAuth_CreateFromConnectionString shall do clean up and return NULL. **] */
                         LogError("mallocAndStrcpy_s failed for sharedAccessKey");
                         free_service_client_auth(result);
                         result = NULL;
                     }
+                    /*Codes_SRS_IOTHUBSERVICECLIENT_12_041: [** IoTHubServiceClientAuth_CreateFromConnectionString shall allocate memory and copy sharedAccessSignature to result->sharedAccessKey by prefixing it with "sas=". **] */
+                    else if (useSharedAccessSignature && (sharedAccess != NULL) && mallocAndStrcpyWithPrefix_s(&result->sharedAccessKey, "sas=", sharedAccess) != 0)
+                    {
+                        /*Codes_SRS_IOTHUBSERVICECLIENT_12_029: [** If the mallocAndStrcpy_s fails, IoTHubServiceClientAuth_CreateFromConnectionString shall do clean up and return NULL. **] */
```

Comment:
>IoTHubServiceClientAuth_CreateFromConnectionString [](start = 100, length = 50)

Update method name to IoTHubServiceClientAuth_CreateFromSharedAccessSignature

Reply:
ok